src/config/local.rs

```diff
@@ -0,0 +1,100 @@
+use {
+    super::Configuration,
+    crate::error::Result,
+    dotenv::dotenv,
+    relay_rpc::domain::ProjectId,
+    serde::Deserialize,
+    std::net::{IpAddr, Ipv4Addr, SocketAddr},
+    url::Url,
+};
+
+// Configuration entrypoint for `cargo run`
+
+#[derive(Deserialize, Debug)]
+pub struct LocalConfiguration {
+    pub project_id: ProjectId,
+    pub registry_auth_token: String,
+
+    #[serde(default = "default_bind_ip")]
```

Comment:
If we're gonna have a custom default for every single field, wouldn't it be easier to just implement `Default`?

Reply:
Not quite sure what you mean: You can't implement default for e.g. IpAddr, and you can't implement it for `bind_ip` so I assume you mean for the whole struct. But then you'd have to merge the configs somehow, and that would require making all these fields optionals? While almost every field has a default value for localhost, it's still nice to allow the dev to override common fields such as port or bind address. This seems like the easiest way to do that.